File: site/src/lib/roles.ts
import type { Role } from "../types";

export function isGlobalAdmin(role?: Role | string | null): boolean {
  return role === "adm";
}

export function isDev(role?: Role | string | null): boolean {
  return role === "dev";
}

export function canManageUsers(role?: Role | string | null): boolean {
  return role === "adm" || role === "adm_local";
}

export function canManageConfig(role?: Role | string | null): boolean {
  return role === "adm";
}

export function canSeeAcesso(role?: Role | string | null): boolean {
  return role === "dev";
}

export function canSeeFilas(role?: Role | string | null): boolean {
  return role === "dev";
}

export function canSeeLogs(_role?: Role | string | null): boolean {
  return true;
}

export function roleLabel(role?: Role | string | null): string {
  switch (role) {
    case "adm":
      return "Admin";
    case "adm_local":
      return "Adm local";
    case "dev":
      return "Dev";
    default:
      return "Usuário";
  }
}

export function creatableRoles(actor?: Role | string | null): Role[] {
  if (actor === "adm") return ["usuario", "adm_local", "adm", "dev"];
  if (actor === "adm_local") return ["usuario", "adm_local"];
  return [];
}

File: api/services/web_api/roles.py
"""Regras de papel do painel.

Roles:
- adm: administrador global (todas as unidades, config)
- adm_local: administrador da unidade (usuarios/destinatarios/logs do estab)
- usuario: operador da unidade
- dev: operacional + auditoria de acessos e filas RabbitMQ
"""

from __future__ import annotations

from typing import Literal

RoleName = Literal["adm", "adm_local", "usuario", "dev"]

ALL_ROLES: tuple[str, ...] = ("adm", "adm_local", "usuario", "dev")
ROLES_COM_ESTABELECIMENTO = frozenset({"adm_local", "usuario"})
ROLES_SEM_ESTABELECIMENTO = frozenset({"adm", "dev"})


def is_global_admin(role: str | None) -> bool:
    return role == "adm"


def is_dev(role: str | None) -> bool:
    return role == "dev"


def is_adm_local(role: str | None) -> bool:
    return role == "adm_local"


def can_manage_users(role: str | None) -> bool:
    return role in ("adm", "adm_local")


def can_manage_config(role: str | None) -> bool:
    return role == "adm"


def can_see_acesso(role: str | None) -> bool:
    return role == "dev"


def can_see_filas(role: str | None) -> bool:
    return role == "dev"


def can_see_logs(role: str | None) -> bool:
    return role in ALL_ROLES


def requires_estabelecimento(role: str | None) -> bool:
    return role in ROLES_COM_ESTABELECIMENTO


def creatable_roles_by(actor_role: str | None) -> tuple[str, ...]:
    if actor_role == "adm":
        return ALL_ROLES
    if actor_role == "adm_local":
        return ("usuario", "adm_local")
    return ()
